hypemaths/models/matrix.py

```diff
@@ -30,6 +33,80 @@ def dims(self) -> tuple:
     def __repr__(self) -> str:
         return f"{self.__class__.__name__}({self.matrix})"
 
+    def multiply_by_number(self, number: t.Union[int, float]) -> 'Matrix':
+        '''
+        Multiplies the current matrix with a number.
+        '''
+        if type(number) is not int:
+            raise TypeError(f"Expected `int` but got {type(number).__name__} ({number})")
+
+        original_matrix = self.matrix
+        new_matrix = Matrix(dims=[original_matrix.rows, original_matrix.cols], fill=0)
+
+        for row_i in range(original_matrix.rows):
+            for column_i in range(original_matrix.cols):
+                new_matrix[row_i][column_i] = number * original_matrix[row_i][column_i]
+
+        return new_matrix
+
+    def transpose(self) -> 'Matrix':
+        '''
+        Transposes the matrix
+        '''
+        # [A.cols, A.rows] is the dimension of the new matrix
+        new_matrix = Matrix(dims=[self.cols, self.rows], fill=0)
+
+        for row_i in range(self.rows):
+            for column_i in range(self.cols):
+                new_matrix[column_i][row_i] = self[row_i][column_i]
```

Comment:
```suggestion
        for row in range(self.rows):
            for column in range(self.cols):
                new_matrix[column][row] = self[row][column]
```
A cleaner approach 😉